aptos-move/framework/aptos-framework/sources/Table.move

```diff
@@ -1,152 +1,108 @@
+    /// table, and cannot be discovered from it.
+    public fun add<K, V>(table: &mut Table<K, V>, key: &K, val: V) {
+        add_box<K, V, Box<V>>(table, key, Box{val});
+        table.length = table.length + 1
     }
 
-    public fun contains_key<Key: store, Value: store>(
-        table: &Table<Key, Value>,
-        key: &Key,
-    ): bool {
-        Option::is_some(&find(table, key))
+    /// Acquire an immutable reference to the value which `key` maps to.
+    /// Aborts if there is no entry for `key`.
+    public fun borrow<K, V>(table: &Table<K, V>, key: &K): &V {
+        &borrow_box<K, V, Box<V>>(table, key).val
     }
 
-    public fun destroy_empty<Key: store, Value: store>(table: Table<Key, Value>) {
-        let Table { data } = table;
-        Vector::destroy_empty(data);
+    /// Acquire a mutable reference to the value which `key` maps to.
+    /// Aborts if there is no entry for `key`.
+    public fun borrow_mut<K, V>(table: &mut Table<K, V>, key: &K): &mut V {
+        &mut borrow_box_mut<K, V, Box<V>>(table, key).val
     }
 
-    public fun insert<Key: store, Value: store>(
-        table: &mut Table<Key, Value>,
-        key: Key,
-        value: Value,
-    ) {
-        let maybe_idx = find(table, &key);
-        assert!(Option::is_none(&maybe_idx), Errors::invalid_argument(EKEY_ALREADY_EXISTS));
-        Vector::push_back(&mut table.data, TableElement { key, value });
+    /// Returns the length of the table, i.e. the number of entries.
+    public fun length<K, V>(table: &Table<K, V>): u64 {
+        table.length
     }
 
-    public fun remove<Key: store, Value: store>(
-        table: &mut Table<Key, Value>,
-        key: &Key,
-    ): (Key, Value) {
-        let maybe_idx = find(table, key);
-        assert!(Option::is_some(&maybe_idx), Errors::invalid_argument(EKEY_NOT_FOUND));
-        let idx = *Option::borrow(&maybe_idx);
-        let TableElement { key, value } = Vector::swap_remove(&mut table.data, idx);
-        (key, value)
+    /// Returns true if this table is empty.
+    public fun empty<K, V>(table: &Table<K, V>): bool {
+        table.length == 0
     }
 
-    fun find<Key: store, Value: store>(
-        table: &Table<Key, Value>,
-        key: &Key,
-    ): Option::Option<u64> {
-        let size = Vector::length(&table.data);
-        let idx = 0;
-        while (idx < size) {
-            if (&Vector::borrow(&table.data, idx).key == key) {
-                return Option::some(idx)
-            };
-            idx = idx + 1
+    /// Acquire a mutable reference to the value which `key` maps to.
+    /// Insert the pair (`key`, `default`) first if there is no entry for `key`.
+    public fun borrow_mut_with_default<K, V: drop>(table: &mut Table<K, V>, key: &K, default: V): &mut V {
+        if (!contains(table, key)) {
+            add(table, key, default)
         };
-
-        Option::none()
+        borrow_mut(table, key)
     }
 
-    #[test]
```

Comment:
why you hate tests? they should still work, right?

Reply:
This file is a copy from upstream, exact same file, and it's tested in a separate unit test file upstream.